test/components/user/list.spec.js

```diff
@@ -1,43 +1,238 @@
+        .respondWithData(() =>
+          testData.standardUsers.sorted().map(testData.toActor))
+        .afterResponses(component => {
+          const { currentUser } = component.vm.$store.state.request.data;
+          for (const tr of component.find('table tbody tr')) {
+            const select = tr.first('select');
+            const { disabled, title } = select.element;
+            const td = tr.first('td');
+            const isCurrentUser = td.text() === currentUser.email;
+            disabled.should.equal(isCurrentUser);
+            (title !== '').should.equal(isCurrentUser);
           }
-        });
+        }));
+
+    describe('clicking the refresh button', () => {
+      it('refreshes the list of users', () =>
+        mockRoute('/users').testRefreshButton({
+          collection: testData.standardUsers,
+          respondWithData: [
+            () => testData.standardUsers.sorted(),
+            () => testData.standardUsers.sorted().map(testData.toActor)
+          ]
+        }));
+
+      it('disables the role selects while the refresh is in progress', () =>
+        mockHttp()
+          .mount(UserList)
+          .respondWithData(() => testData.standardUsers.createPast(1).sorted())
+          .respondWithData(() =>
+            testData.standardUsers.sorted().map(testData.toActor))
+          .complete()
+          .request(component => trigger.click(component, '.btn-refresh'))
+          .beforeEachResponse(component => {
+            for (const select of component.find('table select'))
+              select.getAttribute('disabled').should.equal('disabled');
+          })
+          .respondWithData(() => testData.standardUsers.sorted())
+          .respondWithData(() =>
+            testData.standardUsers.sorted().map(testData.toActor)));
     });
 
-    it('refreshes after the refresh button is clicked', () =>
-      mockRoute('/users')
-        .testRefreshButton({ collection: testData.administrators }));
+    describe('changing a role', () => {
+      const loadUsersAndChangeRole =
+        ({ rowIndex, selectValue, route = false }) =>
+          (route ? mockHttp().mount(UserList) : mockRoute('/users'))
+            .respondWithData(() => testData.standardUsers
+              .createPast(1, {
+                displayName: 'Person 1',
+                email: 'b@email.com',
+                role: 'admin'
+              })
+              .createPast(1, {
+                displayName: 'Person 2',
+                email: 'c@email.com',
+                role: 'none'
+              })
+              .sorted())
+            .respondWithData(() =>
+              testData.toActor(testData.standardUsers.sorted().slice(0, 2)))
+            .complete()
+            .request(component => {
+              const select = component.find('table select')[rowIndex];
+              if (select.element.value === selectValue)
+                throw new Error('no change');
+              select.element.value = selectValue;
```

Comment:
Instead of setting `value` on the `<select>`, is it better to set `selected` on the `<option>`? (We have a similar line to this in the `ProjectUserList` tests as well.)